src/emailcontactproxymodel.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2020 - 2021 Open Mobile Platform LLC.
+ *
+ * This program is licensed under the terms and conditions of the
+ * Apache License, version 2.0.  The full text of the Apache License is at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+#ifndef EMAILCONTAXTPROXYMODEL_H
+#define EMAILCONTACTPROXYMODEL_H
+
+#include <QRegExp>
+#include <QSortFilterProxyModel>
+
+
+class Q_DECL_EXPORT EmailContactProxyModel : public QSortFilterProxyModel
+{
+    Q_OBJECT
+    Q_PROPERTY(int count READ count NOTIFY countChanged)
+    Q_PROPERTY(QString filter READ filter WRITE setFilter NOTIFY filterChanged)
+    Q_PROPERTY(int maxCount READ maxCount WRITE setMaxCount NOTIFY maxCountChanged)
+
+public:
+    explicit EmailContactProxyModel(QObject *parent = nullptr);
+    ~EmailContactProxyModel() override = default;
+
+    QHash<int, QByteArray> roleNames() const override;
+    bool filterAcceptsRow(int source_row, const QModelIndex &source_parent = QModelIndex()) const override;
+    virtual int count() const;
+    QString filter() const;
+    int maxCount() const;
+
+public:
+    void setFilter(const QString &filter);
+    void setMaxCount(const int maxCount);
+
+signals:
+    void countChanged(int);
+    void filterChanged(const QString&);
+    void maxCountChanged(int);
```

Comment:
Minor thing, but with the model being private, i.e. only for qml, the signal parameters don't provide much.